#!/usr/bin/env python
PACKAGE = "plane_segment"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


segment_enum = gen.enum([ gen.const("LineBased",          int_t, 0, ""),
                        gen.const("RANSAC",        int_t, 1, ""),
                        gen.const("Organized",int_t, 2, ""),
                        gen.const("RegionGrow",  int_t, 3, ""),
                        gen.const("AllMethod",  int_t, 4, "")
                       ], "Method to segment plane.")

##
gen.add("segment_method", int_t, 0, "Method to segment plane", 0, 0, 4, edit_method=segment_enum)
##
gen.add("display_input_cloud",  bool_t, 0,  "", True)
gen.add("display_line_cloud",  bool_t, 0,  "", True)
gen.add("display_normal",   bool_t, 0,  "", True)
gen.add("display_plane",    bool_t, 0,  "", True)
gen.add("loop_one_message", bool_t, 0,  "", False)


exit(gen.generate(PACKAGE, "plane_segment", "PlaneSegment"))
